### PR TITLE
Fix runner non return

### PR DESCRIPTION
**Pull request by kiorky**
If the called function crashes with systemexit, no further event is sent, blocking then the awaitening event waiter function in sync mode.

### CODE REVIEW

**Comment by cachedout** [comment]
I encountered this as well. There also seems to be a hiccup in the daemonization call that causes it to hang also. I'll have a PR in shortly.

This looks good, btw, @kiorky. Many thanks!

**Comment by kiorky** [comment]
@cachedout if you are going to to major changes, @jacksontj said to me that he is also on is way, for you to know

**Comment by cachedout** [comment]
@kiorky Very minor, and not in the runner code itself. I don't think I'll conflict with @jacksontj .

**Comment by jacksontj** [comment]
@kiorky I can't seem to cherry pick any of your commits in for some reason, and it no longer merges. I'll just pull in the 2 patches

File: salt/client/mixins.py [hunk 1]
```diff
@@ -231,12 +232,10 @@ def low(self, fun, low):
...
-        except (Exception, SystemExit) as exc:
-            data['return'] = 'Exception occurred in {0} {1}: {2}'.format(
-                            self.client,
-                            fun,
-                            traceback.format_exc(),
-                            )
+        except (Exception, SystemExit):
+            trace = traceback.format_exc()
+            data['return'] = 'Exception occurred in {0} {1}:\n{2}'.format(
+                self.client, fun, trace)
```
Comment (kiorky) [inline salt/client/mixins.py:237]: Exception is printed in the traceback, no need to put it in title !

**Comment by kiorky** [comment]
.. again not applied.
@cachedout i resign ....

**Comment by kiorky** [comment]
Well,

**Pull request closed without merging**